packages/@headlessui-react/src/hooks/use-transition.ts

```diff
@@ -68,98 +68,92 @@ export function useTransition(
 
   let d = useDisposables()
 
-  useIsoMorphicEffect(
-    function retry() {
-      if (!enabled) return
+  useIsoMorphicEffect(() => {
+    if (!enabled) return
 
+    if (show) {
+      setVisible(true)
+    }
+
+    if (!element) {
       if (show) {
-        setVisible(true)
+        addFlag(TransitionState.Enter | TransitionState.Closed)
       }
+      return
+    }
 
-      let node = elementRef.current
-      if (!node) {
-        // Retry if the DOM node isn't available yet
-        if (show) {
-          addFlag(TransitionState.Enter | TransitionState.Closed)
-          return d.nextFrame(() => retry())
```

Comment:
No need for a retry anymore where we _hoped_ the `ref` would be filled in the next time. Now, we just rely on the effect re-running due to a re-render and the dependency array changing.